### ISSUE TITLE
React Native WebRTC randomly crashes if audio track is enabled

### ISSUE THREAD

**Issue by vuminhkh** [react-native-webrtc/react-native-webrtc#1556]
Hello,

I'm testing react native webrtc on ios simulator, and i'm facing a weird error, if I only enable receiving video track then it works fine, but if I enable in the same time audio track then it crashes most of the time (only audio track also crashes). Some times it might work and I have fine video and audio but I just need to terminate the application and restart it again then the issue resurfaces.

```javascript

const peerConnection = new RTCPeerConnection();
  const offerOptions = {
    offerToReceiveAudio: true,
    offerToReceiveVideo: true,
  };
  const offer = await peerConnection.createOffer(offerOptions);
  await peerConnection.setLocalDescription(offer);
  const offer_sdp = offer.sdp!;

  let webRtcAnswer: WebRtcAnswer;
  try {
    webRtcAnswer = await sendWebRTCOffer(connection, cameraId, offer_sdp);
  } catch (err: any) {
    console.error('Failed to start WebRTC stream: ' + err.message);
    peerConnection.close();
    throw err;
  }
  const remoteStream = new MediaStream();
  peerConnection.addEventListener('track', (event: any) => {
    remoteStream.addTrack(event.track);
  });

  const remoteDesc = new RTCSessionDescription({
    type: 'answer',
    sdp: webRtcAnswer.answer,
  });
  try {
    await peerConnection.setRemoteDescription(remoteDesc);
  } catch (err: any) {
    console.error('Failed to connect to WebRTC stream: ' + err.message);
    peerConnection.close();
    throw err;
  }
```
#### Expected Behavior:
It should work with audio track

#### Observed Behavior:
It crashes most of the time to receive audio track,
<img width="1352" alt="Screenshot 2024-04-27 at 03 03 59" src="https://github.com/react-native-webrtc/react-native-webrtc/assets/6248367/5e2acdc7-d21a-48f9-99ba-b8cbd4c02881">

#### Steps to reproduce the issue:
Just run this code on a simulator, most of the time it will fail, remove the audio track then it works fine

#### Platform Information  
* **React Native Version**:  0.74.0
* **WebRTC Module Version**: 118.0.7
* **Platform OS + Version**: iOS 17.2

**Comment by saghul** [comment]
Generally speaking the simulator is not very useful for WebRTC applications on iOS.

The camera is not emulated, no CallKit, and random crashes.

I recommend you test on a real device.

**Comment by vuminhkh** [comment]
Thanks, you are right, it works a lot better on my iphone. I tested in the past a version of my code on my iphone with react native 0.73.5 (it also crashed) but not retested since I upgraded to 0.74.0. I don't remember if there was much difference between the two versions of my code, and I just assumed that it still won't work neither on my iphone with the upgrade to 0.74.0, but I was wrong. Thanks a lot, you saved me days of debugging.
